File: testClients/ruski/AI.py
#-*-python-*-
from BaseAI import BaseAI
from GameObject import *

import random

import game_history
import game_cache
import path_find

class AI(BaseAI):

  manhattan_offsets = [(0,1),(0,-1),(1,0),(-1,0)]

  CLAW, ARCHER, REPAIRER, HACKER, TURRET, WALL, TERMINATOR, HANGAR = range(8)

  """The class implementing gameplay logic."""
  @staticmethod
  def username():
    return "Russley Shaw"

  @staticmethod
  def password():
    return "password"

  ##This function is called once, before your first turn
  def init(self):
    self.history = game_history.game_history(self, True)
    self.cache = game_cache.game_cache(self)
    self.finder = path_find.path_find(self, self.cache)
    pass

  ##This function is called once, after your last turn
  def end(self):
    self.history.print_history()
    pass

  ##This function is called each time it is your turn
  ##Return true to end your turn, return false to ask the server for updated information
  def run(self):
    return self.runSMART()

  def runSMART(self):
    snapshot = self.history.save_snapshot()
    #self.history.print_snapshot(snapshot)
    self.cache.update_all()



    return 1

  def runRAND(self):
    snapshot = self.history.save_snapshot()
    self.history.print_snapshot(snapshot)
    self.cache.update_all()

    #DROP SOMETHING
    randx = random.randrange(0, self.mapWidth-1)
    randy = random.randrange(0, self.mapHeight-1)
    randv = random.randrange(0, 7)
    self.players[self.playerID].orbitalDrop(randx, randy, randv)

    for droid in self.droids:
      if droid.owner == self.playerID:
        offset = random.choice(self.manhattan_offsets)
        droid.move(droid.x + offset[0], droid.y + offset[1])

        offset = random.choice(self.manhattan_offsets)
        droid.operate(droid.x + offset[0], droid.y + offset[1])

    
    return 1

  def __init__(self, conn):
    BaseAI.__init__(self, conn)

File: testClients/HermanAI/AI.cpp
#include "AI.h"
#include "util.h"
#include <iostream>
#include "string"
using namespace std;
enum
{
  CLAW = 0,
  ARCHER = 1,
  REPAIRER = 2,
  HACKER = 3,
  TURRET = 4,
  WALL = 5,
  TERMINATOR = 6,
  HANGAR = 7,
};

AI::AI(Connection* conn) : BaseAI(conn) {}

const char* AI::username()
{
  return "Shell AI";
}

const char* AI::password()
{
  return "password";
}

//This function is run once, before your first turn.
void AI::init(){
  spawnX=5;
  spawnY=1;
}

//This function is called each time it is your turn.
//Return true to end your turn, return false to ask the server for updated information.
bool AI::run()
{
  spawnBitches();
  return true;
}

void AI::spawnBitches()
{
  int unitType=TURRET;
  if(players[playerID()].scrapAmount() >= modelVariants[TURRET].cost())
  {
    if(getTile(spawnX, spawnY)->turnsUntilAssembled() == 0)
    {
      bool spawn = true;
      //make sure there isn't a hangar there
      for(int i = 0; i < droids.size(); i++)
      {
        //if the droid's x and y is the same as the spawn point
        if(droids[i].x() == spawnX && droids[i].y() == spawnY)
        {
          //if the droid is a hangar
          if(droids[i].variant() == HANGAR)
          {
            //can't spawn on top of hangars
            spawn = false;
            break;
          }
        }
      }
      if(spawn)
      {
        //spawn the claw
        players[playerID()].orbitalDrop(spawnX, spawnY, TURRET);
      }
    }
  }
}

//This function is run once, after your last turn.
void AI::end(){
  cout<<"WINNING!!"<<endl;
}

Tile* AI::getTile(const int x, const int y)
{
    if(0 <= x && x < mapWidth() && 0 <= y && y < mapHeight())
        return &tiles[x * mapHeight() + y];
    else
        return NULL;
}




